packages/fab/index.js

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+export default class Fab extends React.Component {
+
+  state = {
+    classList: new Set(),
+  };
+
+  get classes() {
+    const {classList} = this.state;
+    const {
+      mini,
+      className,
+    } = this.props;
+
+    return classnames('mdc-fab', Array.from(classList), className, {
+      'mdc-fab--mini': mini,
+    });
+  }
+
+  setClassOnChildren = () => {
+    return React.Children.map(this.props.children, (item) => {
+      const className = `${item.props.className} mdc-fab__icon`;
+      const props = Object.assign({}, item.props, {className});
+      return React.cloneElement(item, props);
+    });
+  };
+
+  render() {
+    const children = this.setClassOnChildren();
+    return (
+        <button className={this.classes}>
+          {children}
+        </button>
+    );
+  }
+}
+
+Fab.propTypes = {
+  mini: PropTypes.bool,
+  className: PropTypes.string,
+  children: PropTypes.oneOfType([
```

Comment:
I think Fab's propTypes should be similar to TopAppBar's propTypes, which allowed our clients to pass in their icons, or a MaterialIcon

Reply:
So pass it in using a prop, or continue to use a child element? The reason we used a prop for top app bar was the icons had to be placed 2 `div` elements down (inside a section that was within a row). Since this is such a simple element, does it make sense to make to use a prop when a direct child element does the same thing?

Reply:
I guess I just dont understand the difference between PropTypes.node and PropTypes.element. Why does TopAppBar have elements, and Fab has nodes?

In TopAppBar we have this code:
```
  addClassesToElement(classes, element) {
    const propsWithClasses = {
      className: classnames(classes, element.props.className),
    };
    return React.cloneElement(element, propsWithClasses);
  }
```

Maybe we should pull that out of top app bar so we can use it in multiple components

Reply:
FYI, `element` is more restrictive than `node`, IIUC. From prop-types' README:

```
  // Anything that can be rendered: numbers, strings, elements or an array
  // (or fragment) containing these types.
  optionalNode: PropTypes.node,

  // A React element.
  optionalElement: PropTypes.element,
```

Reply:
We chose element since we need it to be an element that we can add a class to. As a FAB, it should always be an element (<a>, <i>, <svg>, etc). 

We may have issues when we get to extended Fab, but with regards to Fab `element` is what we want.